api/src/certification/configuration/infrastructure/repositories/flash-algorithm-configuration-repository.js

```diff
@@ -1,6 +1,38 @@
+/**
+ * @typedef {import ('../../../shared/domain/models/FlashAssessmentAlgorithmConfiguration.js').FlashAssessmentAlgorithmConfiguration} FlashAssessmentAlgorithmConfiguration
+ */
+
 import { DomainTransaction } from '../../../../shared/domain/DomainTransaction.js';
 
+/**
+ * @param {FlashAlgorithmConfiguration} flashAlgorithmConfiguration
+ * @returns {Promise<void>}
+ */
 export const save = async (flashAlgorithmConfiguration) => {
   const knexConn = DomainTransaction.getConnection();
-  return knexConn('flash-algorithm-configurations').insert(flashAlgorithmConfiguration.toDTO());
+
+  const currentActiveConfiguration = await knexConn('certification-configurations')
```

Comment:
Proposition de renomage de cette variable pour faire comprendre qu'il faut absolument faire ce SELECT avant INSERT


```suggestion
  const existingActiveConfigurationBeforeInsert = await knexConn('certification-configurations')
```